lib/bap_disasm/bap_disasm_rec.ml

```diff
@@ -213,26 +203,75 @@ let next dis s =
 
 let stop_on = [`Valid]
 
-(* find intersections, i.e. instructions which share bytes, but have different address,
-   and update destinations for those of them, which also share max address to
-   split a block in such place *)
+module Dest = struct
+  let rec compare_edge e e' =
+    if e = e' then 0
+    else
+      match e, e' with
+      | `Fall, _ -> -1
+      | `Jump, _ -> 1
+      | _ -> - (compare_edge e' e)
+
+  let compare (d,e) (d',e') =
```

Comment:
can you, please, explain the reasoning behind this ordering? 

Both here, and leave the comment in the code.